### DISCUSSION TITLE
Custom Fields into PDF template

### DISCUSSION THREAD

**Discussion by igjunior** [kimai/Paid-plugins#93] [Questions & Answers] [answered]
I would love to be able to get data from my custom fields in the timesheet into a custom PDF export.

I've tried so many variations of something like this but nothing seems to work. Any ideas how to access the data into the twig template?

`{% set customField = timesheet.getCustomField('fieldname') %}`
`{{ customField }}`

**Comment by kevinpapst** [comment]
Almost:
```twig
{% set customField = timesheet.getCustomField('fieldname') %}
{{ customField.getValue() }}
```

**Reply by igjunior** [reply]
I must be doing something wrong since it's still not working with that code. Is there anything else that I might need to do? 
Otherwise I'll just figure out another way to do what I'm trying to accomplish.  

<img width="421" alt="image" src="https://github.com/kevinpapst/kimai-plugins/assets/143045963/dcb45436-054e-40c2-88ff-a320cebf741e">
<img width="535" alt="image" src="https://github.com/kevinpapst/kimai-plugins/assets/143045963/c2635cd3-ccd5-4e75-be62-7a63383e8d32">

**Reply by kevinpapst** [reply, marked as answer]
Sorry, I didn't check because your code looked good 🤷 

I added documentation at 
https://www.kimai.org/documentation/export.html#pdf-templates
```
                            {% set field = entry.metaField(key) %}
                            {% if field is not null and field.value is not null %}
                                    {{ field.value }}
                            {% endif %}
```

**Reply by igjunior** [reply]
Perfect - that was it! Next mission is to figure out how to sum down all values from a custom field in the query, I'll post the result if that works out. Thanks again!
